src/pages/index.js

```diff
@@ -10,70 +10,133 @@ import PopupWithForm from '../components/PopupWithForm.js';
+      handleDeleteClick: (cardId, cardElement) => {
+        deleteCardPopup.open(cardId, cardElement);
+      },
+      handleLikeClick: (cardId, isLiked, cardInstance) => {
+        if (isLiked) {
+          api.unlikeCard(cardId)
+            .then(updatedCardData => {
+              cardInstance.updateLikes(updatedCardData.likes);
+            })
+            .catch(err => console.error(err));
+        } else {
+          api.likeCard(cardId)
+            .then(updatedCardData => {
+              cardInstance.updateLikes(updatedCardData.likes);
+            })
+            .catch(err => console.error(err));
+        }
+      }
     }
   );
   return card.getView();
 }
 
+
+const deleteCardPopup = new PopupWithConfirmation('#delete-confirm-modal', {
+  handleFormSubmit: (cardId, cardElement) => {
+    if (!cardId) {
+      cardElement.remove();
+      deleteCardPopup.close();
+      return;
+    }
+    api.deleteCard(cardId)
+    .then(() => {
+      cardElement.remove();
+      deleteCardPopup.close();
+    })
+    .catch(err => console.error(err));
+}
+});
+
+deleteCardPopup.setEventListeners();
+
+
 const cardSection = new Section(
   {
-    items: initialCards,
+    items: [], 
     renderer: (cardItem) => {
       const cardElement = createCard(cardItem);
       cardSection.addItem(cardElement);
     }
   },
   '.cards__list'
 );
+
+
 cardSection.renderItems();
 
-const editProfilePopup = new PopupWithForm(
-  '#profile-edit-modal',
-  (formData) => {
-    userInfo.setUserInfo({
-      name: formData.title,
-      description: formData.description
+const editProfilePopup = new PopupWithForm('#profile-edit-modal', (formData) => {
+  return api.setUserInfo(formData.title, formData.description)
+    .then((updatedUserData) => {
+      userInfo.setUserInfo({
+        name: updatedUserData.name,
+        description: updatedUserData.about,
+        avatar: updatedUserData.avatar 
+      });
     });
-  }
-);
+});
 editProfilePopup.setEventListeners();
 
-const addCardPopup = new PopupWithForm(
-  '#profile-add-card',
-  (formData) => {
-    const newCard = createCard({
-      name: formData.title,
-      link: formData.url
+const addCardPopup = new PopupWithForm('#profile-add-card', (formData) => {
+  return api.addNewCard(formData.title, formData.url)
+    .then((newCardData) => {
+      const newCardElement = createCard(newCardData);
+      cardSection.addItem(newCardElement);
     });
-    cardSection.addItem(newCard);
-
-    addCardFormValidator.disableButton();
-  }
-);
+});
 addCardPopup.setEventListeners();
 
+
+const avatarPopup = new PopupWithForm('#avatar-edit-modal', (formData) => {
+  return api.setAvatar(formData.avatar)
+    .then((updatedUser) => {
+      userInfo.setUserInfo({
+        name: updatedUser.name,
+        description: updatedUser.about,
+        avatar: updatedUser.avatar
+      });
+    });
+});
+
+avatarPopup.setEventListeners();
+avatarContainer.addEventListener('click', () => {
+   avatarPopup.open();
+  });
+
```

Comment:
* Please, create the `avatar` form **validator**, otherwise vilidation of the `avatar` form doesn't work